build/openresty/patches/ngx_lua-0.10.25_01-dyn_upstream_keepalive.patch

```diff
@@ -1,8 +1,20 @@
+diff --git a/bundle/nginx-1.21.4/src/http/ngx_http_upstream.c b/bundle/nginx-1.21.4/src/http/ngx_http_upstream.c
+index b07e564..a72dcf2 100644
+--- a/bundle/nginx-1.21.4/src/http/ngx_http_upstream.c
++++ b/bundle/nginx-1.21.4/src/http/ngx_http_upstream.c
+@@ -4304,6 +4304,7 @@ ngx_http_upstream_next(ngx_http_request_t *r, ngx_http_upstream_t *u,
+     if (u->peer.cached && ft_type == NGX_HTTP_UPSTREAM_FT_ERROR) {
+         /* TODO: inform balancer instead */
+         u->peer.tries++;
++        u->peer.notify(&u->peer, u->peer.data, 0);
```

Comment:
Now that the balancer has been notified, `u->peer.tries++;` is not needed anymore as implied by the comment line `/* TODO: inform balancer instead */`. Just let the balancer decide what to do.

Reply:
`u->peer.tries++;` is the original code in nginx, and I am concerned that deleting it may affect the logic of other balancers in the original code. So I am considering only notifying `ngx_http_lua_balancer` because it is a custom implementation.